packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1360,11 +1360,6 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
-  void _hideSelectionOverlayIfNeeded() {
```

Comment:
I think some of the problem came from the fact that there are two subtly different methods with confusing names: `_hideSelectionOverlayIfNeeded` and `hideToolbar`.  I got rid of this one altogether and I think it makes more sense now.